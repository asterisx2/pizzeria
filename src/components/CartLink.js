import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';

export default class CartLink extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      quantity: this.props.cart.length
    };
  }

  static propTypes = {
    store: PropTypes.shape({
      pizzas: PropTypes.arrayOf(
        PropTypes.shape({
          id: PropTypes.number.isRequired,
          size: PropTypes.string.isRequired,
          basePrice: PropTypes.number.isRequired,
          quantity: PropTypes.number.isRequired,
          toppings: PropTypes.arrayOf(
            PropTypes.shape({
              __typename: PropTypes.string.isRequired,
              name: PropTypes.string.isRequired,
              price: PropTypes.number.isRequired
            }).isRequired
          ).isRequired
        }).isRequired
      ).isRequired
    })
  };

  getDerivedStateFromProps(nextProps, prevState) {
    if (nextProps.total !== prevState.total) {
      if (nextProps.store.pizzas.length > 0) {
        let sum = 0;
        nextProps.store.pizzas.map(pizza => (sum += pizza.quantity));
        return { quantity: sum };
      }
    }
  }

  render() {
    return (
      <div>
        <Link 
          to="/cart" 
          className="text-light">
          <i 
            className="fa fa-shopping-cart pr-1" 
            aria-hidden="true" />
            ({this.state.quantity})
        </Link>
      </div>
    );
  }
}
