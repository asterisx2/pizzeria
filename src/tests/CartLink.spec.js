import React from 'react';
import CartLink from '../components/CartLink';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';

describe('CartLink Component', () => {
  let cart = [];

  it('renders without crashing', () => {
    const tree = renderer.create(
            <MemoryRouter>
                <CartLink
                    cart={cart}
                />
            </MemoryRouter>
        ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
