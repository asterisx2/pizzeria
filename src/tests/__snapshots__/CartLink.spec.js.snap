// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartLink Component renders without crashing 1`] = `
<div>
  <a
    className="text-light"
    href="/cart"
    onClick={[Function]}
  >
    <i
      aria-hidden="true"
      className="fa fa-shopping-cart pr-1"
    />
    (
    0
    )
  </a>
</div>
`;
